internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -326,3 +350,43 @@ func (r *NumaflowControllerRolloutReconciler) SetupWithManager(mgr ctrl.Manager)
 		For(&apiv1.NumaflowControllerRollout{}).WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// SplitYAMLToString splits a YAML file into strings. Returns list of yamls
+// found in the yaml. If an error occurs, returns objects that have been parsed so far too.
+func SplitYAMLToString(yamlData []byte) ([]string, error) {
```

Comment:
maybe later we can add a unit test for some of these? should we add a todo for it?